tools/ot/src/test/java/dk/alexandra/fresco/tools/ot/otextension/CheatingNetwork.java

```diff
@@ -1,27 +1,31 @@
 package dk.alexandra.fresco.tools.ot.otextension;
 
 import dk.alexandra.fresco.framework.configuration.NetworkConfiguration;
-import dk.alexandra.fresco.framework.network.KryoNetNetwork;
+import dk.alexandra.fresco.framework.network.CloseableNetwork;
+import dk.alexandra.fresco.framework.network.async.AsyncNetwork;
+import java.io.IOException;
 
 /**
  * Wrapper class used for testing to ensure that proper checked exceptions are
  * thrown if a party is trying to cheat. This class intercepts messages send
  * over the network and flips a bit when asked to.
  */
-public class CheatingNetwork extends KryoNetNetwork {
+public class CheatingNetwork implements CloseableNetwork {
+  private final CloseableNetwork network;
+
   private int cheatByteNo;
   private int messageNo;
   private int counter = 0;
   private boolean cheat = false;
 
   public CheatingNetwork(NetworkConfiguration conf) {
-    super(conf);
+    this.network = new AsyncNetwork(conf);
```

Comment:
Since it now has become a decorator, then call the network decorated or delegate

Reply:
Good point ... in fact, think to should not even be dependent on AsyncNetwork. We could use any closeable network here.